local-macros/src/lib.rs

```diff
@@ -117,33 +123,86 @@ pub fn ast_type(attr: TokenStream, item: TokenStream) -> TokenStream {
                 }
             }
         }
-        Enum(DataEnum { variants, .. }) => {
-            let variant_names = variants.iter().map(|variant| &variant.ident);
-            let variant_names_2 = variant_names.clone();
-            let variant_names_3 = variant_names.clone();
-            let variant_names_4 = variant_names.clone();
-            let variant_names_5 = variant_names.clone();
-            let variant_names_6 = variant_names.clone();
-            let variant_names_7 = variant_names.clone();
-            let variant_names_8 = variant_names.clone();
-            let variant_names_9 = variant_names.clone();
-            let variant_names_10 = variant_names.clone();
-            let variant_names_11 = variant_names.clone();
-            let variant_names_12 = variant_names.clone();
-            let variant_names_13 = variant_names.clone();
-            let variant_names_14 = variant_names.clone();
+        "NamedDeclarationInterface" => {
+            quote! {
+                impl crate::NamedDeclarationInterface for #ast_type_name {
+                    fn name(&self) -> ::std::rc::Rc<crate::Node> {
```

Comment:
What does the `::` in front of `std` mean?

Reply:
I think it's sort of like a leading `/` in an absolute path, it's saying "this is a root name, not a relative/local name" (or whatever the right terminology is)

Like in Rails if you were to say `::ModelName` to disambiguate the model from a module-local class named `Module::ModelName`